admiral/pkg/clusters/virtualservice_handler.go

```diff
@@ -397,49 +427,62 @@ func syncVirtualServiceToRemoteCluster(
 	remoteRegistry *RemoteRegistry,
 	virtualService *v1alpha3.VirtualService,
 	event common.Event,
-	syncNamespace string) error {
+	syncNamespace string,
+	vSName string) error {
 
 	ctxLogger := log.WithFields(log.Fields{
 		"type":     "syncVirtualServicesToAllRemoteClusters",
-		"identity": virtualService.Name,
+		"identity": vSName,
 		"txId":     uuid.New().String(),
 	})
 
 	defer logElapsedTimeForVirtualService("syncVirtualServiceToRemoteCluster="+string(event), cluster, virtualService)()
 	rc := remoteRegistry.GetRemoteController(cluster)
 	if rc == nil {
-		return fmt.Errorf(LogFormat, "Event", common.VirtualServiceResourceType, virtualService.Name, cluster, "remote controller not initialized for cluster")
+		return fmt.Errorf(LogFormat, "Event", common.VirtualServiceResourceType, vSName, cluster, "remote controller not initialized for cluster")
 	}
 	if rc.VirtualServiceController == nil {
-		return fmt.Errorf(LogFormat, "Event", common.VirtualServiceResourceType, virtualService.Name, cluster, "VirtualService controller not initialized for cluster")
+		return fmt.Errorf(LogFormat, "Event", common.VirtualServiceResourceType, vSName, cluster, "VirtualService controller not initialized for cluster")
 	}
+
 	if event == common.Delete {
-		err := rc.VirtualServiceController.IstioClient.NetworkingV1alpha3().VirtualServices(syncNamespace).Delete(ctx, virtualService.Name, metav1.DeleteOptions{})
+		// Best effort delete for existing virtual service with old name
+		_ = rc.VirtualServiceController.IstioClient.NetworkingV1alpha3().VirtualServices(syncNamespace).Delete(ctx, virtualService.Name, metav1.DeleteOptions{})
```

Comment:
Why not return an err from here as well?

Reply:
Doing best effort as this will eventually always fail once all the vs's are overwritten with new name.